setup.cfg

```diff
@@ -53,3 +53,13 @@ exclude = (?x)(
   | dask_cudf/dask_cudf/tests/
   # This close paren cannot be in column zero otherwise the config parser barfs
  )
+
+[codespell]
+skip = ./.git,./.github,./cpp/build,.*egg-info.*,versioneer.py,./.mypy_cache,./cpp/tests,./python/cudf/cudf/tests
+ignore-words-list = inout,indext,offsett,dout,unparseable,incase,afterall,fiter,trings
```

Comment:
This list is not that long! 💯

Reply:
I'm cheating a bit and ignoring the tests =) 

(The tests have some test data in there that was messing it up - but wasn't really typos)

Reply:
I think that’s reasonable. We can always expand later if we wish. The errors fixed in this PR are worthwhile even if not covering the whole repo.

Reply:
You probably could add `./java/src/test` to this list and then remove `trings`
And add `./cpp/include/cudf_test/cxxopts.hpp` and then remove `fiter`.
I'm not finding `indext` or `afterall` in the code.
Is `offsett` in here because of `offsetted`? Maybe that can be corrected.

Reply:
Also, should the `skip` list include `./python/cudf/_skbuild` and `./python/strings_udf/_skbuild` as well?

Reply:
It doesn't seem like its possible to get codespell to automatically exclude files from .gitignore  yet =( https://github.com/codespell-project/codespell/issues/1887 

Running codespell through pre-commit won't run on any files that aren't being tracked by git though, which should get around having to mirror the .gitignore file.   I've changed the codespell pre-commit exclude regex to exclude test files and versioneer.py files - which should make codespell pass when run with `pre-commit run --all-files`

The skips in setup.cfg are for running codespell in its interactive mode, and will be ignored when used with pre-commit - because of basically the same issue in https://github.com/rapidsai/cudf/pull/12036  (pre-commit passes an explicit list of files to codespell, which the `skip` section provided here doesn't override).  We could perhaps entirely remove this skip section , if we don't want to support running directly codespell from the command line rather than from pre-commit.

I've also added java/src/test to the skips, and removed trings, afterall and indext.

`offsett` is because of template params like https://github.com/rapidsai/cudf/blob/7d173c9d144a64c5e1a0467d2a5eb4181854f25e/cpp/src/io/fst/agent_dfa.cuh#L79-L84

Reply:
Can we fix the `fiter` one? It only appears in `cxxopts.hpp` and seems like a typo:
https://github.com/rapidsai/cudf/blob/4497ed6bcd2347ba7a77f4b8d9e7f867c10e1d42/cpp/include/cudf_test/cxxopts.hpp#L1417-L1434

Reply:
> `fitr` would work, right

I would be inclined to just call this `iter` or `it` ?

Reply:
> `offsett` is because of template params like

Add `\b[a-zA-Z][[:wordlike:]]T\b` to the ignore regex?

Reply:
I believe `cxxopts.hpp` is vendored and should be ignored rather than edited. I believe that specific name `fiter` is supposed to mean “format iterator” and is not a typo.

Reply:
Then it should not be linted at all.
If you look at the history for `cxxopts.hpp` it has been modified it several times.
I realize `fiter` is not a typo but it seems odd to create an exception for it for the entire repo.
Either we should not lint this file or we should fix it.

Reply:
I agree with David that we should not lint this file. (My previous comment may have been unclear.)

Reply:
I've removed the cxxopts.hpp from linting - and applied @wence- suggestion about the ignore regex to exclude template params types. The ignore-words list is down to just `inout,unparseable` now, and I think those are both valid words myself.

Reply:
Perfect! That's ideal.